reports/report-manager.js

```diff
@@ -0,0 +1,226 @@
+      deleteCallback({message: 'Report not found.'});
+      return;
+    }
+
+    // If the report's student UID does not match the user UID, and the user is
+    // not an admin, 'report' should not be deleted.
+    // TODO(jmtaber129): Check report's student UID and user UID, and add error
+    // handling.
+
+    report.remove(deleteCallback);
+  });
+};
+
+/**
+ * Finds a specific report, and calls a callback once the report is found.
+ * @param {string} reportUid - The UID corresponding to the report to be found.
+ * @param {Object} locals - An object containing the current request's local
+ *     variables.
+ * @param {reportCallback} queryCallback - Called once the operation finishes.
+ */
+var getReportById = (reportUid, locals, queryCallback) => {
+  Report.findById(reportUid, (err, report) => {
+    if (err) {
+      queryCallback(err);
+      return;
+    }
+
+    // If the report's student UID does not match the user UID, and the user is
+    // not an admin, 'report' should not be returned.
+    // TODO(jmtaber129): Check report's student UID and user UID, and add error
+    // handling.
+
+    queryCallback(err, report);
+  });
+};
+
+/**
+ * Callback used by report CRUD methods that return multiple reports.
+ * @callback multipleReportsCallback
+ * @param {Error} err - An error that occurred during the operation.  Null if no
+ *     errors occurred.
+ * @param {Object.<string, Report>} reports - An object of reports with their
+ *     UIDs as keys.
+ */
+
+/**
+ * Finds all reports, and calls a callback once all reports are found.
+ * @param {Object} reqData - An object containing additional search parameters
+ *     (not yet implemented).
+ * @param {Object} locals - An object containing the current request's local
+ *     variables.
+ * @param {multipleReportsCallback} queryCallback - Called once the operation
+ *     finishes.
+ */
+var getAllReports = (reqData, locals, queryCallback) => {
+  conditions = {};
+
+  // If the user is not an admin, limit the results to reports with the user's
+  // UID.  If the user is not an admin, and one of the query parameters is a
+  // student UID other than their own, handle it as an error.
+  // TODO(jmtaber129): Set conditions.student to user UID for non-admins.
+  // TODO(jmtaber129): Add error handling for requesting reports with student
+  // UIDs other than the UID of a non-admin user.
+
+  // TODO(jmtaber129): Set additional conditions once query parameters are
+  // accepted in the request object.
+
+  // Both report types are in the same collection, and querying a model queries
+  // the entire collection, so querying one report model will return results for
+  // both report types.
+  Report.find(conditions, (err, reports) => {
+    if (err) {
+      queryCallback(err);
+      return;
+    }
+
+    returnObject = {};
+
+    reports.forEach((report) => {
+      returnObject[report.id] = report;
+
+      // TODO(jmtaber129): Add point value to report in 'returnObject'.
```

Comment:
Nah, the TODO is fine. I was I was just commenting further to keep anyone reading this in the loop as to why point value is _not_ yet being dealt with.